src/base/bittorrent/private/filterparserthread.cpp

```diff
@@ -28,16 +28,73 @@
  * Contact : chris@qbittorrent.org
  */
 
-#include <QFile>
-#include <QHostAddress>
+#include "filterparserthread.h"
+
 #include <QDataStream>
+#include <QFile>
 #include <QStringList>
 
 #include "base/logger.h"
-#include "filterparserthread.h"
 
 namespace libt = libtorrent;
 
+namespace
+{
+    class DATIPv4Parser {
+    public:
+        bool tryParse(const char* str) {
+            unsigned char number = 0;
+
+            const char* numberStart = str;
+            char* endptr;
+            for (; *str; ++str) {
+                if (*str == '.') {
+                    long int extractedNum = strtol(numberStart, &endptr, 10);
+                    if ((extractedNum >= 0L) && (extractedNum <=255L))
+                        m_buf[number++] = static_cast<unsigned char>(extractedNum);
+                    else
+                        return false;
+
+                    if (endptr != str)
+                        return false;
+                    if (number == 4)
+                        return true;
+
+                    numberStart = str + 1;
+                }
+            }
+
+            if (str != numberStart) {
+                m_buf[number] = static_cast<unsigned char>(strtol(numberStart, &endptr, 10));
+                if (endptr == str && number == 3)
+                    return true;
+            }
+
+            return false;
+        }
+
+        libt::address_v4::bytes_type parsed() const {
+            return m_buf;
+        }
+
+    private:
+        libt::address_v4::bytes_type m_buf;
+    };
+
+    bool parseIPAddress(const QByteArray &data, libt::address &address)
```

Comment:
I couldn't figure out a way to make strtol (and address_v6::from_string) to look only in a portion of the buffer.